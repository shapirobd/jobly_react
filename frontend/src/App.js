import React from "react";
import Routes from "./Routes";
import NavBar from "./NavBar";
import "./App.css";
import UserContext from "./UserContext";
import useLocalStorage from "./useLocalStorage";

function App() {
	// const [user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
	// const [token, setToken] = useState(JSON.parse(localStorage.getItem("token")));
	// const [isLoggedIn, setIsLoggedIn] = useState(false);
	// const [loggingOut, setLoggingOut] = useState(false);

	const {
		user,
		setUser,
		token,
		setToken,
		isLoggedIn,
		setIsLoggedIn,
		loggingOut,
		setLoggingOut,
		isUpdated,
		setIsUpdated,
	} = useLocalStorage();

	return (
		<div className="App">
			<UserContext.Provider
				value={{
					user,
					setUser,
					token,
					setToken,
					isLoggedIn,
					setIsLoggedIn,
					loggingOut,
					setLoggingOut,
					isUpdated,
					setIsUpdated,
				}}
			>
				<NavBar />
				<Routes />
			</UserContext.Provider>
		</div>
	);
}

export default App;
