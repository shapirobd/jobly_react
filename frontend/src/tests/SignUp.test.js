import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "../App";

test("smoke test", () => {
	render(
		<MemoryRouter initialEntries={["/signup"]}>
			<App />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={["/signup"]}>
			<App />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
