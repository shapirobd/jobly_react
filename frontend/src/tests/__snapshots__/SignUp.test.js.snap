// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="NavBar"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <a
                class="makeStyles-title-6"
                href="/"
                variant="h6"
              >
                Jobly
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="NavBar-routes"
          >
            <a
              href="/login"
            >
              Login
            </a>
            <a
              href="/signup"
            >
              Sign up
            </a>
          </div>
        </div>
      </header>
    </div>
    <div
      class="SignUp"
    >
      <h4
        class="MuiTypography-root SignUp-heading MuiTypography-h4"
      >
        Sign Up
      </h4>
      <form
        class="SignUp-form"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username-input"
          name="username"
          type="text"
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password-input"
          name="password"
          type="password"
        />
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          id="firstName-input"
          name="firstName"
          type="text"
        />
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          id="lastName-input"
          name="lastName"
          type="text"
        />
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email-input"
          name="email"
          type="email"
        />
        <div
          class="SignUp-btn-wrapper"
        >
          <button
            class="SignUp-btn"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
